### PR TITLE
include/cpp-tree: add missing ostream

### PR DESCRIPTION
**Pull request by wjwithagen**
FreeBSD/stdlib requires ostream:
/home/jenkins/workspace/ceph-master/src/include/cpp-btree/btree.h:2537:10: error: invalid operands to binary expression ('std::ostream' (aka 'basic_ostream<char>') and 'const char [3]')
      os << "  ";
      ~~ ^  ~~~~
/usr/include/c++/v1/type_traits:4034:3: note: candidate function template not viable: cannot convert argument of incomplete type 'std::ostream' (aka 'basic_ostream<char>') to 'std::byte' for 1st argument
  operator<< (byte  __lhs, _Integer __shift) noexcept
  ^
/usr/include/c++/v1/memory:5264:1: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-2>' against 'char const[3]'
operator<<(basic_ostream<_CharT, _Traits>& __os, shared_ptr<_Yp> const& __p);
^
/usr/include/c++/v1/string:4244:1: note: candidate template ignored: could not match 'basic_string<type-parameter-0-0, type-parameter-0-1, type-parameter-0-2>' against 'char const[3]'
operator<<(basic_ostream<_CharT, _Traits>& __os,
^
1 error generated.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
i am removing the dump methods in #32692

File: src/include/cpp-btree/btree.h [hunk 1]
```diff
@@ -55,6 +55,7 @@
 #include <iterator>
 #include <limits>
 #include <new>
+#include <ostream>
```
Comment (tchaikov) [inline src/include/cpp-btree/btree.h]: if we really want to have `operator<<` for btree. could just remove `iosfwd` couple lines above as `ostream` has the complete type definition for ostream. while iosfwd just offers the forward declaration for it.

Comment (wjwithagen) [reply src/include/cpp-btree/btree.h]: @tchaikov FreeBSD seems happy with that.

**Comment by tchaikov** [comment]
closing in favor of #32692

**Pull request closed without merging**